plugin/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+cmake_minimum_required(VERSION 3.25.0 FATAL_ERROR)
+project(QOfonoQtDeclarative LANGUAGES CXX)
+
+include(GNUInstallDirs)
+
+set(PLUGIN_SOURCES
+    qofonodeclarativeplugin.cpp
+    qofononetworkoperatorlistmodel.cpp
+    qofonosimlistmodel.cpp
+)
+
+add_library(QOfonoQtDeclarative SHARED ${PLUGIN_SOURCES})
+
+target_include_directories(QOfonoQtDeclarative PRIVATE
+    ${CMAKE_SOURCE_DIR}/src
+)
+
+target_link_libraries(QOfonoQtDeclarative PRIVATE
+    Qt5::Qml
+    Qt5::DBus
+    qofono-qt${QT_MAJOR_VERSION}
+)
+
+install(TARGETS QOfonoQtDeclarative
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/qt5/qml/QOfono
+)
+
+install(FILES qmldir plugins.qmltypes
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/qt5/qml/QOfono
+)
+
+add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/plugins.qmltypes
+    COMMAND /bin/sh -c "qmlplugindump -nonrelocatable QOfono 0.2 > ${CMAKE_CURRENT_SOURCE_DIR}/plugins.qmltypes"
```

Comment:
this `qmlplugindump` command failed on my desktop with `QQmlComponent: Component is not ready` and iirc this was the reason why we ended up manually generating these `plugins.qmltypes` files (it failed in obs too or something like that)

Reply:
I've reverted generation of `plugin.qmltypes` which is now manual again as suggested in https://github.com/sailfishos/libqofonoext/pull/10#issuecomment-3673321050